src/prefect/tasks.py

```diff
@@ -1508,7 +1517,7 @@ async def serve(self) -> NoReturn:
         """
         from prefect.task_worker import serve
 
-        await serve(self)
+        asyncio.run(serve(self))
```

Comment:
this, with removing `sync_compatible` from the `serve` task utility, gives us clean Ctrl+C exits for `task.serve`